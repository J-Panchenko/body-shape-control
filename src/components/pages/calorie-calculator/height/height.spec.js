import { render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import React from 'react';
import Height from './height';

const setup = () => {
  const utils = render(<Height />);
  const input = utils.getByLabelText('height-input');
  return {
    input,
    ...utils,
  };
};

describe('Height component: positive cases', () => {
  const { input } = setup();

  it('It should allow numbers to be inputted', () => {
    fireEvent.change(input, { target: { value: '190' } });
    expect(input.value).toBe('190');
  });

  it('It should not allow letters to be inputted', () => {
    fireEvent.change(input, { target: { value: 'My height' } });
    expect(input.value).toBe('');
  });

  it('It should allow numbers from 60 to 260 to be inputted', () => {
    fireEvent.change(input, { target: { value: '12' } });
    expect(input.value).toBe('12');
    expect(input.validity.stepMismatch).toBe(false);

    fireEvent.change(input, { target: { value: '380' } });
    expect(input.value).toBe('380');
    expect(input.validity.stepMismatch).toBe(false);
  });

  it('matches snapshot', () => {
    const renderer = create(<Height />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
