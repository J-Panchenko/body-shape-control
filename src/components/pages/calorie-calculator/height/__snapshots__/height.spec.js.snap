// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Height component: positive cases matches snapshot 1`] = `
<label
  className="parameter__item"
  htmlFor="height"
>
  <h2
    className="parameter-subtitle"
  >
    Height
  </h2>
  <span>
    cm
  </span>
  <input
    aria-label="height-input"
    className="input"
    id="height"
    max={260}
    min={60}
    placeholder="0"
    required={true}
    step={0.1}
    type="number"
  />
</label>
`;
